rooms/windyTylne.cpp

```diff
@@ -1,24 +1,214 @@
+#include <cstdlib>
 #include <iostream>
+#include <string>
 
 #include "../main.hpp"
 
-Room windyTylne(__attribute__((unused)) GameData &gameData) {
-  std::cout << "Jesteś w windach tylnych" << std::endl
-            << "Czy chcesz iść do biblioteki?" << std::endl;
-  if (yes()) {
-    return BIBLIOTEKA;
-  }
-  std::cout << "Czy chcesz iść do garażu?" << std::endl;
-  if (yes()) {
-    return GARAZ;
-  }
-  std::cout << "Czy chcesz iść do windy na piętrze?" << std::endl;
-  if (yes()) {
-    return WINDA_PIETRO;
+const int8_t minigame_row = 3;
+const int8_t minigame_col = 3;
+
+void start(GameData& gameData) {
+  std::cout << "Przechodzisz do tylnej strony budynku i widzisz "
+            << (gameData.winda_repaired
+                    ? "naprawioną przez siebę windę "
+                    : "windę, która sprawia wrażenie niedziałającej")
+            << std::endl;
+}
+/*
+void draw_minigame(char k[][minigame_col+2], short off[minigame_col]) {
+    std::cout << "Not yet implemented" << std::endl;
+}
+*/
+void input_prompt(std::string& s) {
+  std::cout << "> ";
+  std::cin >> s;
+}
+
+bool lift_repairing_game(GameData& gameData) {  // TODO(seqre): Add minigame
+  std::string choice;
+  std::cout << "Po chwili dochodzisz do wniosku, że bez ściągnięcia"
+            << "zewnętrznej osłony nic nie zdziałasz" << std::endl;
+  if (gameData.inventory.find("śrubokręt") != gameData.inventory.end()) {
+    std::cout
+        << "Odkręcasz śrubki trzymające ochronną płytkę i ostrożnie ją ściągasz"
+        << " podważając śrubokrętem" << std::endl
+        << "Widzisz porwane kable bez izolacji, które czasami się stykając"
+        << "tworzą efektowne zwarcia" << std::endl
+        << "Co robisz?" << std::endl
+        << "1. Naprawiam dalej" << std::endl
+        << "2. Odpuszczam i przykręcam płytkę z powrotem" << std::endl;
+
+    input_prompt(choice);
+    if (choice[0] == '1') {
+      std::cout << "Not implemented yet" << std::endl;
+      /*
+      std::cout << "Po chwili analizowania kabli stwierdzasz, że jak je
+      poprzesuwasz i odpowiednio połączysz"
+                << " to przyciski powinny zacząć działać" << std::endl;
+
+      bool kable_poprawnie = false;
+      bool ruch = false;
+
+      char kable [minigame_row][minigame_col+2] {};
+      short offset [minigame_col] {};
+
+      while(kable_poprawnie) {
+          while(!ruch) {
+              clear_screen();
+              draw_minigame(kable, offset);
+          }
+
+      }
+      */
+      return false;
+    }
+    return false;
   }
-  std::cout << "Czy chcesz iść do windy na parterze?" << std::endl;
-  if (yes()) {
-    return WINDA_PARTER;
+    std::cout << "Niestety, ale nie masz potrzebnego narzędzia do tego, wróć "
+                 "później jak je zdobędziesz"
+              << std::endl;
+    return false;
+}
+
+Room choose_room(GameData& gameData, int8_t inside_idiot_counter) {
```

Comment:
this function is long and hard to read, split it.

Reply:
Done